src/main/java/com/hazelcast/gcp/GcpDiscoveryStrategyFactory.java

```diff
@@ -50,4 +59,69 @@ public DiscoveryStrategy newDiscoveryStrategy(DiscoveryNode discoveryNode, ILogg
         }
         return result;
     }
+
+    /**
+     * Checks if Hazelcast is running on GCP.
+     * <p>
+     * To check if Hazelcast is running on GCP, we first check whether the internal DNS is configured for "google.internal" in
+     * either "/etc/resolv.conf" or "/etc/hosts". Such an approach is not officially documented but seems like a good
+     * enough heuristic to detect a GCP Compute VM Instance. Since it's not the official method, we still need to make
+     * an API call to "metadata.google.internal" which will resolve to a local, non-routable address http://169.254.169.254/.
+     * Finally, we check if there is a service account attached for this instance because without a service account Hazelcast
+     * GCP discovery will not work.
+     *
+     * @return true if running on GCP Instance which has a service account attached
+     * @see https://cloud.google.com/compute/docs/instances/managing-instances#dmi
+     */
+    @Override
+    public boolean isAutoDetectionApplicable() {
+        return googleInternalDnsConfigured() && metadataFlavorGoogle() && serviceAccountAttached();
+    }
+
+    private static boolean googleInternalDnsConfigured() {
+        return readFileContents("/etc/resolv.conf").contains("google.internal")
+                || readFileContents("/etc/hosts").contains("google.internal");
+    }
+
+    static String readFileContents(String fileName) {
+        InputStream is = null;
+        try {
+            File file = new File(fileName);
+            byte[] data = new byte[(int) file.length()];
+            is = new FileInputStream(file);
+            is.read(data);
+            return new String(data, StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            throw new RuntimeException("Could not get " + fileName, e);
+        } finally {
+            IOUtil.closeResource(is);
+        }
+    }
+
+    private static boolean metadataFlavorGoogle() {
+        return isEndpointAvailable("metadata.google.internal");
```

Comment:
This part does not work for me. I get the following exception.
```
com.hazelcast.gcp.RestClientException: Failure in executing REST call
        at com.hazelcast.gcp.RestClient.call(RestClient.java:98)
        at com.hazelcast.gcp.RestClient.get(RestClient.java:61)
        at com.hazelcast.gcp.GcpDiscoveryStrategyFactory.isEndpointAvailable(GcpDiscoveryStrategyFactory.java:135)
        at com.hazelcast.gcp.GcpDiscoveryStrategyFactory.metadataFlavorGoogle(GcpDiscoveryStrategyFactory.java:113)
        at com.hazelcast.gcp.GcpDiscoveryStrategyFactory.isAutoDetectionApplicable(GcpDiscoveryStrategyFactory.java:78)
        at com.hazelcast.spi.discovery.impl.DefaultDiscoveryService.detectDiscoveryStrategyFactory(DefaultDiscoveryService.java:207)
        at com.hazelcast.spi.discovery.impl.DefaultDiscoveryService.loadDiscoveryStrategies(DefaultDiscoveryService.java:147)
        at com.hazelcast.spi.discovery.impl.DefaultDiscoveryService.<init>(DefaultDiscoveryService.java:58)
        at com.hazelcast.spi.discovery.impl.DefaultDiscoveryServiceProvider.newDiscoveryService(DefaultDiscoveryServiceProvider.java:29)
        at com.hazelcast.instance.impl.Node.createDiscoveryService(Node.java:339)
        at com.hazelcast.instance.impl.Node.<init>(Node.java:265)
        at com.hazelcast.instance.impl.HazelcastInstanceImpl.createNode(HazelcastInstanceImpl.java:148)
        at com.hazelcast.instance.impl.HazelcastInstanceImpl.<init>(HazelcastInstanceImpl.java:117)
        at com.hazelcast.instance.impl.HazelcastInstanceFactory.constructHazelcastInstance(HazelcastInstanceFactory.java:211)
        at com.hazelcast.instance.impl.HazelcastInstanceFactory.newHazelcastInstance(HazelcastInstanceFactory.java:190)
        at com.hazelcast.instance.impl.HazelcastInstanceFactory.newHazelcastInstance(HazelcastInstanceFactory.java:128)
        at com.hazelcast.core.Hazelcast.newHazelcastInstance(Hazelcast.java:91)
        at com.hazelcast.core.server.HazelcastMemberStarter.main(HazelcastMemberStarter.java:47)
Caused by: java.net.MalformedURLException: no protocol: metadata.google.internal
        at java.base/java.net.URL.<init>(URL.java:645)
        at java.base/java.net.URL.<init>(URL.java:541)
        at java.base/java.net.URL.<init>(URL.java:488)
        at com.hazelcast.gcp.RestClient.call(RestClient.java:72)
```

Could you double check?

Reply:
Can you tell me what you ran for this to happen?

Reply:
Yeah, I just created a new GCP VM, copied there Hazelcast and Hazelcast Kubernetes plugin Jars and executed:
```
java -cp hazelcast-kubernetes.jar:hazelcast.jar com.hazelcast.core.server.HazelcastMemberStarter
```

Also you need to enabled debug logging or change the log level in the code.

Reply:
Do you mean gcp plug-ins?

Reply:
yeah, sorry, I meant `hazelcast-gcp.jar`